### DISCUSSION TITLE
第1回目バックエンドチーム打ち合わせ 議事録

### DISCUSSION THREAD

**Discussion by ghost** [Progaku-copy/progaku-archive#54] [General]
### 開発について
- レビューはりくやさんへ
  - ただ、他のメンバーで気づいたことがあればコメントしてもらっていい 
- swaggerでメモのAPI確認できる
- 現状はメモのcrudのみ
- コメントはまだ作成してないかも
- テストはrspecで下記を作成している
  - モデルスペック
  - リクエストスペック
- Openapiの仕様書もフロントのために作成した方がいいと思っている
- gem committee を導入済みなので、定義した API とテスト対象の API response が一致しているかを確認してくれるメソッドが使用できるようになっている
- テストカバレッジ100%を目指すためにgem simplecovの導入オススメ
- アプリの完成期限は10月中

### 稼働時間の目安
- Nine　4-12h/週
- タロス　6h/週
- たかくさぎ　10-15/週
- りくや　10-16/週
- nova　8h/週
※ Nineさんはフロントのデザイン関連タスク優先になると思う

### 要検討事項
- アカウント認証
  - Progaku専用にするのかしないのか
  - アカウント認証はdevise使う？
  - googleのauth認証？
- ページネーション
  - 実装必須だが、gem kaminariじゃない方がいいかも

### Q&A
Q：自分で足りないと思った機能があった際など、Issueはりくやさん以外も作っていい？
A：作っていいです！ただ、完成時期を守ることは意識して、りくやさんに事前確認は必要

Q：環境整備の効率化どうする？
A：issue作った時に自動でprojectで未着手に入る等の効率化はした方がいいかも。Gituhubのラベル管理とかいいかも（「手が空いたら」「緊急」等）

Q：デプロイ先は決まっている？
A：未定。おすすめのサービスあったら教えて欲しい

**Comment by ghost** [comment]
ところどころ間違っている箇所があるかもしれませんので、気づいた方は修正お願いします🙇‍♂️

**Comment by kuri0616** [comment]
> ~Openapiの仕様書もフロントのために作成した方がいいと思っている~
打ち合わせの結果、openapi仕様書は今後使用しない方向でいく。スプレッドシートで再度作成する。
理由 = キャッチアップに工数を要するため。モックサーバ、コード自動生成等使用せず、参照するだけならスプレッドシートで記載した方が、認知負荷が低いため

少しだけ補足追加しました！
